executor/include/executor/best_fit.hpp

```diff
@@ -79,7 +88,12 @@ void enable_exec_with_priority(
     }
     return false;
   });
-  // TODO: Throw warning or assert
+
+
+  // This should not happen, at least one executor should always be selected. So either all runtime checks failed which is
+  // incorrect behaviour or no supported executors were passed which should not be done
+  if (!executor_selected)
+      std::cerr<<"No executor selected. All runtime checks returned false or no executors were passed."<<std::endl;
```

Comment:
I'm surprised this is not resolved already at compile time.

Reply:
Well if you have no run-time checks then it is known at compile time whether or not some executor will be selected but adding runtime checks takes away that.